linker/shared/src/main/scala/org/scalajs/linker/frontend/optimizer/OptimizerCore.scala

```diff
@@ -2082,7 +2082,17 @@ private[optimizer] abstract class OptimizerCore(
             else dynamicCall(className, methodName)
           if (impls.size == 1) {
             pretransformSingleDispatch(flags, impls.head, Some(treceiver), targs, isStat, usePreTransform)(cont) {
-              treeNotInlined
+              if (isWasm) {
+                // Replace by an ApplyStatically to guarantee static dispatch
+                val targetClassName = impls.head.enclosingClassName
+                val castTReceiver = foldCast(treceiver, ClassType(targetClassName))
+                cont(PreTransTree(ApplyStatically(flags,
+                    finishTransformExprMaybeAssumeNotNull(castTReceiver),
+                    targetClassName, methodIdent,
+                    targs.map(finishTransformExpr))(resultType), RefinedType(resultType)))
+              } else {
+                treeNotInlined
```

Comment:
Just out of curiosity, why we don't apply this change to JS (generates `ApplyStatically`)?

Reply:
[Previous experiments showed that it was not a win, or even detrimental, in JS.](https://github.com/scala-js/scala-js/pull/4337)

Reply:
Thanks! That's interesting, maybe it's nice to add a link to that PR for future developers?